interview-questions.md

```diff
@@ -8,50 +8,74 @@ Then, research the question to expand on your answer. Even if you feel you have
+  return chocolate, vanilla
+}
+
 
   Researched answer:
 
+A parameter is the variable notated inside the parenthsis that acts as a placeholder for what is expected to be called back in the function or given value. 
+
+An argument is the value or expression that is sent back to the parameter to run through the function. 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
   Your answer:
 
-  Researched answer:
+The .map() feature is a feature that creates a new array after going through each element in an array after a function has been called. The .map() feature uses the function command that was addressed to go through each element in the array one time and returns a new array with the new values that resulted. 
 
 
+  Researched answer:
+
+The .map() returns a new array with new values that resulted from going through an arrays indivual elements.
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: .map() is a tool that is used to return pieces of an array individually. The .map() feature goes through each portion of an array and is commonly used in loops for iteration purposes. The .filter() returns a subset of an array of wanted elemnts by removing those we did not want. 
 
   Researched answer:
 
+The .map() tool returns indivual elements of an array indivually and returns a new array. The .filter() feature returns a new array by removing those elements we did not want in the function byways of the argument.
 
 
 4. What is the difference between a function and a method?
 
   Your answer:
 
+A function has a set of code that can be called directly by its name with or without paramter indentification.
+
+
+A method is a function you use in an object or class and can be called using square bracket or . notation to call the object.
+
   Researched answer:
 
+A fuction is a set of resuable code that code be used anywhere and it passes data and returns unique data. A method is a function that is specifically called by the name of the object associated. Although they are both the same one is speicific to an object which is a class instance. 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: I like object destructuring becuase it is almost like the unpacking of values from a object properties or arrays. It is almost easy becuase a clear return is usually given when breaking them down.
 
   Researched answer:
 
-
+Object destructing is used to extract data variables from arrays or object properties. This feature allows you to extract multiple pieces of data at once and you can even build new functions out of those datas.
 
 6. STRETCH: What is hoisting in JavaScript?
 
   Your answer:
+I dont know exactly to be honest. But I am assuming it has to do with moving a variable and I Say this becuase hoisting means to move something up. This term sounds familiar but I can not remember.
```

Comment:
Really love this answer!  In an interview setting, you will often encounter questions or terminology that you may be unfamiliar with, and saying you are unsure but relating the word as best as possible to something you can define is a great strategy.  It allows an interviewer to see that you can connect dots and recognize relationships between topics.